### ISSUE TITLE
How to use PairGrid and colors to g.map_diag()

### ISSUE THREAD

**Issue by joelostblom** [mwaskom/seaborn#933]
Hi,
I would like to use `PairGrid` and color the plot in `g.map_diag()` in another color than the default without setting the hue parameter.

When I change the palette parameter in `PairGrid`, nothing happens 

```
iris = sns.load_dataset("iris")
iris = iris[['sepal_width', 'sepal_length', 'species']]
g = sns.PairGrid(iris, palette="Oranges")
g = g.map_offdiag(plt.scatter)
g = g.map_diag(plt.hist)
```

![index](https://cloud.githubusercontent.com/assets/4560057/15455663/1a81f074-2029-11e6-8a32-ecb7e191e022.png)

unless I specify the hue parameter.

```
g = sns.PairGrid(iris, hue='species', palette="Oranges")
g = g.map_offdiag(plt.scatter)
g = g.map_diag(plt.hist)
```

![index2](https://cloud.githubusercontent.com/assets/4560057/15455667/459fb2dc-2029-11e6-955e-987c0535972a.png)

I can then change the color of the scatter plot manually,

```
g = sns.PairGrid(iris, hue='species', palette="Oranges")
g = g.map_offdiag(plt.scatter, color='black')
g = g.map_diag(plt.hist)
```

![index3](https://cloud.githubusercontent.com/assets/4560057/15455686/dc5f2b08-2029-11e6-8328-4ff22faf052d.png)

but when I try the same for the histogram, it throws me an error. I see a similar error if trying `sns.kdeplot` with `g.map_diag`.

```
g = sns.PairGrid(iris, hue='species', palette="Oranges")
g = g.map_offdiag(plt.scatter, color='black')
g = g.map_diag(plt.hist, color='black')
```

```
TypeError: hist() got multiple values for keyword argument 'color'
```

My current workaround for this is to create a new parameter that is the same for all rows and group by that, so that the color is applied as I wish.

```
iris['group'] = 'dummy'
g = sns.PairGrid(iris, hue='group', palette="Oranges")
g = g.map_offdiag(plt.scatter, color='black')
g = g.map_diag(plt.hist)
```

![index34](https://cloud.githubusercontent.com/assets/4560057/15455806/204f605a-202d-11e6-8a83-2044c4a6a1de.png)

Out of coincidence, I noticed that if I assign `iris['group']` the string `'1'` instead of `'dummy'`, I run into an error, although `iris.dtypes` shows me that the column holds objects in both cases.

```
ValueError: First argument must be a sequence
```

Am I misunderstanding how to use `PairGrid`? I don't understand why the color palette is not applied without using the hue parameter and why I can specify a separate color to `g.map_offdiag`, but not to `g.map_diag`.

**Comment by mwaskom** [comment]
I guess that's what https://github.com/mwaskom/seaborn/pull/788 is about

**Comment by mwaskom** [comment]
Closed with #788
